File: Array/zhangjianan/20200717数字转化为十六进制/toHex.c
char * toHex(int num){
   if (num == 0) return "0";
    char *ret;char nums[] ="0123456789abcdef";
    ret = (char *)(calloc(9,sizeof(char )));
    int i = 7, j = 8;
    do {
        int la= num & 0xf;
        ret[i--] = nums[la];
        num >>= 4;
        j--;
    } while (num != 0 && j > 0);
    return ret+i+1;


}

File: Array/zhangjianan/20200805数字转化为十六进制/tohex.java
class Solution {
    public String toHex(int num) {
        StringBuffer    sb=new StringBuffer();
        String s="0123456789abcdef";
        while(num!=0){
            int index=num&15;
            sb.append(s.charAt(index));
            num>>>=4;
        }
        if(sb.length()==0)sb.append(0);
        return sb.reverse().toString();
    }
}